src/field_theory/minpoly.lean

```diff
@@ -52,6 +52,9 @@ by { delta minpoly, rw dif_pos hx, exact (well_founded.min_mem degree_lt_wf _ hx
 lemma ne_zero [nontrivial A] (hx : is_integral A x) : minpoly A x ≠ 0 :=
 ne_zero_of_monic (monic hx)
 
+lemma eq_zero (hx : ¬ is_integral A x) : minpoly A x = 0 :=
+dif_neg hx
```

Comment:
Can you combine these into and `iff`?

Reply:
Nope. Only if `A` is nontrivial, see the lemma above it. Also, I guess we mostly want to use this in the form
```
rw [eq_zero hx]
```
so this form seems best to me.